proposals/js/JsName.md

```diff
@@ -81,6 +88,108 @@ Additionally it can be used to force using functions for accessors to avoid prob
     - *translates an accessing to such property as call to accessor function using name from the annotation.*
 
 
+## How it should work for overrides?
+
+Actually, the problems actual even we prohibit using `JsName` on overrides,
+because the compiler doesn't mangle the name of native declarations,
+so we can simply get name clash when to inherit from two natives.
+
+And doesn't make sense to prohibit to use this annotation on non-final declarations.
+
+So propose to **prohibit using JsName on overrides**, but **report about name clashes** on non-native declarations.
+
+
+**Case 1: declarations with different names from different parents with the same requested name (by JsName)**
+    _(map many names to the one)_
+
+
+_Note: A, B, C can be native._
+
+Example 1:
+```kotlin
+interface A {
+    @JsName("boo")
+    fun bar()
+}
+
+
+interface B {
+    @JsName("boo")
+    fun baz()
+}
+
+class C : A, B {
+    override fun bar() {}
+    override fun baz() {}
+}
+```
+
+Example 2:
+```kotlin
+interface A {
+    @JsName("boo2")
+    fun bar2() {}
+}
+
+
+interface B {
+    @JsName("boo2")
+    fun baz2() {}
+}
+
+class C : A, B
+```
+
+Proposed solution: allow name clashes on native declarations and prohibit on non-native declarations.
+
+**Case 2: declarations with the same name from different parents with different requested name (by JsName):**
+    _(map the one name to many)_
+
+_Note: A, B, C can be native._
+
+Example 1:
+```kotlin
+interface A {
+    @JsName("fooA")
+    fun foo()
+}
+
+
+interface B {
+    @JsName("fooB")
+    fun foo()
+}
+
+class C : A, B {
+    override fun foo() {}
+}
+```
```

Comment:
Consider the following (quite rare) case:

* `A` and `B` are written by different people and distributed in different libraries
* both `A` and `B` are non-native
* both `foo()` declarations have `@JsName` on them (probably, due to authors wishing to resolve some name conflict) with distinct values.

I see no reason to prohibit this. We could generate something like:

```
C.prototype.fooA = function() {
    // translated body of `C.foo`
};
C.prototype.fooB = C.prototype.fooA;
```

There's no matter which of `fooA` and `fooB` to use on a call site when receiver is known to be `C`. When receiver is either `A` or `B` we generate `fooA` and `fooB` respectively.